pkg/resources/core/v1/secret/mutator_test.go

```diff
@@ -363,6 +366,70 @@ func TestMutatorAdmit(t *testing.T) {
 	}
 }
 
+func TestMutatorAdmitOnCreate(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name      string
+		secret    corev1.Secret
+		wantAdmit bool
+		wantErr   bool
+	}{
+		{
+			name:      "create secret",
+			secret:    corev1.Secret{},
+			wantAdmit: true,
+		},
+		{
+			name: "create cloud credential secret",
+			secret: corev1.Secret{
+				Type: "provisioning.cattle.io/cloud-credential",
+			},
+			wantAdmit: true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			req := admission.Request{
+				AdmissionRequest: admissionv1.AdmissionRequest{
+					UID:             "2",
+					Kind:            secretGVK,
+					Resource:        secretGVR,
+					RequestKind:     &secretGVK,
+					RequestResource: &secretGVR,
+					Name:            "my-secret",
+					Namespace:       "test-ns",
+					Operation:       admissionv1.Create,
+					UserInfo:        authenicationv1.UserInfo{Username: "test-user", UID: ""},
+					Object:          runtime.RawExtension{},
+					OldObject:       runtime.RawExtension{},
+				},
+			}
+			var err error
+			req.Object.Raw, err = json.Marshal(test.secret)
+			require.NoError(t, err)
+
+			ctrl := gomock.NewController(t)
```

Comment:
Minor, but I believe you could move all this ctrl portion above the for loop since none of it relies on the tests themselves

Reply:
Sure, I'll move this above the test loop, since each test case here requires the mutator to be constructed in the same way.